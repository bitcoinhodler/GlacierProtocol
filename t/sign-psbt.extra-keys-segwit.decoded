{
  "txid": "90ecfc94c1c6526139cb37818db7393e35725b420ffdd5dd8b87b3ea1806fa03",
  "hash": "2fb2a4325bd433ba4ddb89a2d16c3a618ffd9939cf785c3f524cc7eaa1f9d77c",
  "version": 2,
  "size": 406,
  "vsize": 190,
  "weight": 760,
  "locktime": 0,
  "vin": [
    {
      "txid": "8d1ec0732d14892e97d858e79e88d63c33c381686b5a9f45dd34a9d89612d058",
      "vout": 1,
      "scriptSig": {
        "asm": "00201e6d327472a0dbd0cd64ea03229a86cf865f02029a29ad06635c5e6d83213328",
        "hex": "2200201e6d327472a0dbd0cd64ea03229a86cf865f02029a29ad06635c5e6d83213328"
      },
      "txinwitness": [
        "",
        "304402201ed3d1b3bb64ab8dbb3ac6a02a595839fb922a1ff9e702962330dd5ea888b1d0022065c1f29bc24a3da3eb9eabae117b4787078e537a38127fa514bb3985d7050e2a01",
        "304402203fd6324c183e73c704b9837a24846586acf7005020f1035ef6359e31f7dff3a8022025fd4c2782a37da0fb7a5bfd99feb8e0066f8b766fbd28be68ae3be8199c1d8a01",
        "5221027cac2fe4bb16fa9cdd57de980f546d7cddd8704f11dfc37803ce8831955caf3821031344a608e2a2e7f0a4fd3f60e004f633ef4361482b5bbf4e420a07035ca3554121023d6acb654efdd5b6354dd6bfa2e1fab9feab6468d970fdac9b373dfadb11d5952103a6c60cec3e79f91bec8453f534501f05ad2b0b6e6e1faae05e7a2391d65d998054ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.01998100,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 c6e28c80218f62fc1bbd767d53a1d419fd5a7bb0 OP_EQUAL",
        "hex": "a914c6e28c80218f62fc1bbd767d53a1d419fd5a7bb087",
        "reqSigs": 1,
        "type": "scripthash",
        "addresses": [
          "2NBNqB9wqYYxgcCSeqNrR9RmHc4NsLm7jWe"
        ]
      }
    }
  ]
}
