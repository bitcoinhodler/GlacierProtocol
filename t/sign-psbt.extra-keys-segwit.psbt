cHNidP8BAFMCAAAAAQF9k54xBT37cztvWEvQYUNM4zd+vWGyxodQlXZPUtmEAQAAAAD/////ARR9HgAAAAAAF6kUxuKMgCGPYvwbvXZ9U6HUGf1ae7CHAAAAAAABASCAhB4AAAAAABepFB9KKOTamQkkGBMSB1IQErGc1F8+hwEEIgAgHm0ydHKg29DNZOoDIpqGz4ZfAgKaKa0GY1xebYMhMygBBYtSIQJ8rC/kuxb6nN1X3pgPVG183dhwTxHfw3gDzogxlVyvOCEDE0SmCOKi5/Ck/T9g4AT2M+9DYUgrW79OQgoHA1yjVUEhAj1qy2VO/dW2NU3Wv6Lh+rn+q2Ro2XD9rJs3PfrbEdWVIQOmxgzsPnn5G+yEU/U0UB8FrSsLbm4fquBeeiOR1l2ZgFSuAAA=
