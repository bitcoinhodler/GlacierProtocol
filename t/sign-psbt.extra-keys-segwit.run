#!/bin/bash
set -e

# Test validation of extra keys. Only need 2, providing 4 as I would in the test withdrawal process


$GLACIERSCRIPT --regtest=$1 sign-psbt << INPUT 2>&1
y
y
y
y
y
y
../../t/sign-psbt.extra-keys-segwit.psbt
y
4
cUY5DRXSmpqT7WpBAkjFz6PPxHoPJ2Cw2VNCa7muihpRjAUJVycs
cQTQEE1ooQHpVi5C4aTcbYPSUMzY2gF1DVdafUsEwnYYaT97UUMy
cQsAGjLXgGXpa48W8kUh3Pci8bMDENoUW5zVkanknDUWd4uxQSPh
cTQHU4ZzwLthJhDU6oxSzmWsTmwhDvFphJCfKFDZxxy6F8j37jRq
y
INPUT
