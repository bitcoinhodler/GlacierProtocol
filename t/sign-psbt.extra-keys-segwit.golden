Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?Input a filename located in the current directory which contains the PSBT:
0.02000000 BTC in unspent inputs from cold storage address 2Mv6fmwU6Nj6tiSNHzxSDz8vFiEJkYQ46Fm
0.01998100 BTC going to destination address 2NBNqB9wqYYxgcCSeqNrR9RmHc4NsLm7jWe
Fee amount: 0.00001900
Confirm? (y/n): 
How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: Key #3: Key #4: 
Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.02000000 BTC in unspent inputs from cold storage address 2Mv6fmwU6Nj6tiSNHzxSDz8vFiEJkYQ46Fm
0.01998100 BTC going to destination address 2NBNqB9wqYYxgcCSeqNrR9RmHc4NsLm7jWe
Fee amount: 0.00001900

Signing with private keys: 
cUY5DRXSmpqT7WpBAkjFz6PPxHoPJ2Cw2VNCa7muihpRjAUJVycs
cQTQEE1ooQHpVi5C4aTcbYPSUMzY2gF1DVdafUsEwnYYaT97UUMy
cQsAGjLXgGXpa48W8kUh3Pci8bMDENoUW5zVkanknDUWd4uxQSPh
cTQHU4ZzwLthJhDU6oxSzmWsTmwhDvFphJCfKFDZxxy6F8j37jRq


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 10.00 satoshis per vbyte

Raw signed transaction (hex):
0200000000010158d01296d8a934dd459f5a6b6881c3333cd6889ee758d8972e89142d73c01e8d01000000232200201e6d327472a0dbd0cd64ea03229a86cf865f02029a29ad06635c5e6d83213328ffffffff01147d1e000000000017a914c6e28c80218f62fc1bbd767d53a1d419fd5a7bb087040047304402201ed3d1b3bb64ab8dbb3ac6a02a595839fb922a1ff9e702962330dd5ea888b1d0022065c1f29bc24a3da3eb9eabae117b4787078e537a38127fa514bb3985d7050e2a0147304402203fd6324c183e73c704b9837a24846586acf7005020f1035ef6359e31f7dff3a8022025fd4c2782a37da0fb7a5bfd99feb8e0066f8b766fbd28be68ae3be8199c1d8a018b5221027cac2fe4bb16fa9cdd57de980f546d7cddd8704f11dfc37803ce8831955caf3821031344a608e2a2e7f0a4fd3f60e004f633ef4361482b5bbf4e420a07035ca3554121023d6acb654efdd5b6354dd6bfa2e1fab9feab6468d970fdac9b373dfadb11d5952103a6c60cec3e79f91bec8453f534501f05ad2b0b6e6e1faae05e7a2391d65d998054ae00000000

Transaction fingerprint (md5):
dc475a0edd451bb38bf33310933b1ffb
QR code for transaction written to transaction.png
